File: docker_sdk/python/docker_sdk/model/shared_service.py
from typing import List

from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column, relationship
from docker_sdk.model.base import Base


class SharedService(Base):
    __tablename__ = 'shared_services'

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(30), unique=True)
    engine: Mapped[str] = mapped_column(String(30))
    version: Mapped[str] = mapped_column(String(30))

    projects: Mapped[List["SharedServiceProject"]] = relationship(
        back_populates='shared_service'
    )

File: docker_sdk/php/docker-sdk/src/Model/SharedService.php
<?php

/**
 * This file is part of the Spryker Suite.
 * For full license information, please view the LICENSE file that was distributed with this source code.
 */

namespace DockerSDK\Model;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class SharedService extends Model
{
    protected $table = 'shared_services';

    protected $fillable = [
        'name',
        'engine',
        'version',
    ];

    public $timestamps = false;

    public function projects(): BelongsToMany
    {
        return $this->belongsToMany(
            Project::class,
            'shared_services_projects',
            'shared_service_id',
            'project_id',
        )->using(
            SharedServiceProject::class
        )->withPivot(['data']);
    }
}
